### PR TITLE
db: Add option to capture canceled jobs

### PR DESCRIPTION
**Pull request by V-FEXrt**
Adds `--canceled` which captures jobs that were canceled in the last build (usually by ctrl+c). This works similarly to `--failed` in that it will forget the list of canceled jobs after the next wake invocation.

### CODE REVIEW

**Review by colinschmidt — APPROVED** [review]
LGTM

**Comment by V-FEXrt** [comment]
Need to run a test to verify that wakebox timed out jobs show up as canceled then I can land this

File: tests/inspection/canceled/pass.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#! /bin/sh
+
+set -e
+WAKE="${1:+$1/wake}"
+
+# Start from any empty db every time for stable job ids
+rm -f wake.db
+
+# Use || true to ignore the expected non-0 return from timeout
+timeout 1 ${WAKE} test || true
```
Comment (colinschmidt) [inline tests/inspection/canceled/pass.sh:10]: We don't need `test Unit` here?

File: tools/wake/cli_options.h [hunk 2]
```diff
@@ -139,9 +140,10 @@ struct CommandLineOptions {
       {'g', "globals", GOPT_ARGUMENT_FORBIDDEN},
       {'e', "exports", GOPT_ARGUMENT_FORBIDDEN},
       {0, "html", GOPT_ARGUMENT_FORBIDDEN},
-      {0, "timeline", GOPT_ARGUMENT_OPTIONAL},
-      {0, "simple-timeline", GOPT_ARGUMENT_OPTIONAL},
-      {0, "simple", GOPT_ARGUMENT_OPTIONAL},
+      {0, "timeline", GOPT_ARGUMENT_FORBIDDEN},
+      {0, "simple-timeline", GOPT_ARGUMENT_FORBIDDEN},
+      {0, "simple", GOPT_ARGUMENT_FORBIDDEN},
+      {0, "canceled", GOPT_ARGUMENT_FORBIDDEN},
```
Comment (colinschmidt) [inline tools/wake/cli_options.h:146]: oops. thanks for fixing the other recent additions.

File: tools/wake/main.cpp [hunk 3]
```diff
@@ -428,10 +434,11 @@ int main(int argc, char **argv) {
...
   }
 
-  bool is_db_inspection = !clo.job_ids.empty() || !clo.output_files.empty() ||
-                          !clo.input_files.empty() || !clo.labels.empty() || !clo.tags.empty() ||
-                          clo.last_use || clo.last_exe || clo.failed || clo.tagdag ||
-                          clo.timeline || clo.taguri || clo.simple || clo.simple_timeline;
+  bool is_db_inspection =
+      !clo.job_ids.empty() || !clo.output_files.empty() || !clo.input_files.empty() ||
+      !clo.labels.empty() || !clo.tags.empty() || clo.last_use || clo.last_exe || clo.failed ||
+      clo.tagdag || clo.timeline || clo.taguri || clo.simple || clo.simple_timeline || clo.canceled;
```
Comment (colinschmidt) [inline tools/wake/main.cpp:440]: Do we have a linter for C++?

File: tests/inspection/canceled/pass.sh (hunk 1 above)
Comment (V-FEXrt) [reply tests/inspection/canceled/pass.sh:10]: Nope. `test` is exported as a CLI function so it expects a `List String` and with nothing is specified then `Nil` is passed

File: tools/wake/main.cpp (hunk 3 above)
Comment (V-FEXrt) [reply tools/wake/main.cpp:440]: Yep, just for formatting though. `clang-format` (version 12.0.X)

**Pull request merged**